ad7768-1fmcz/ad77681.c

```diff
@@ -243,10 +313,321 @@ int32_t ad77681_spi_read_adc_data(struct ad77681_dev *dev,
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_CRC_status_handling(struct ad77681_dev *dev,
+	uint16_t *data_buffer)
+{
+	int32_t ret = 0;
+	uint8_t status_byte = 0, checksum, checksum_byte, checksum_buf[5], checksum_length, i;
+	char print_buf[50];
+	
+	// Status bit handling
+	if(dev->status_bit)																			// Status bit attached to SPI data frame
+	{
+		if (dev->conv_len == AD77681_CONV_24BIT)												// 24bit ADC data + 8bit of status = 2 16bit frames
+			status_byte = data_buffer[1] & 0xFF;															
+		else																					// 16bit ADC data + 8bit of status = 2 16bit frames
+			status_byte = data_buffer[1] >> 8;															
+	}
+	
+	// Chcecksum bit handling
+	if(dev->crc_sel != AD77681_NO_CRC)
+	{
+		if (dev->status_bit)
+		{
+			if (dev->conv_len == AD77681_CONV_24BIT)											// 24bit ADC data + 8bit of status + 8bit of CRC = 3 16bit frames
+				{
+					checksum_byte = data_buffer[2] >> 8;
+					checksum_length = 4;
+				}		
+			else																				// 16bit ADC data + 8bit of status + 8bit of CRC = 2 16bit frames
+				{
+					checksum_byte = data_buffer[1] & 0xFF;
+					checksum_length = 3;
+				}	
+		}
+		
+		else																					// without status bit
+			{
+				if (dev->conv_len == AD77681_CONV_24BIT)											// 24bit ADC data + 8bit of CRC = 2 16bit frames
+					{
+						checksum_byte = data_buffer[1] & 0xFF;
+						checksum_length = 3;
+					}
+				
+				else																				// 16bit ADC data + 8bit of CRC = 2 16bit frames
+					{
+						checksum_byte = data_buffer[1] >> 8;
+						checksum_length = 2;
+					}	
+			}
+		
+		for (i = 0; i < checksum_length; i++)
+		{
+			if (i % 2)
+				checksum_buf[i] = data_buffer[i / 2] & 0xFF;
+			else
+				checksum_buf[i] = data_buffer[i / 2] >> 8;
+		}
+		
+		if (dev->crc_sel == AD77681_CRC)
+			checksum = ad77681_compute_crc8(checksum_buf, checksum_length, INITIAL_CRC_CRC8);
+		
+		else if (dev->crc_sel == AD77681_XOR)
+			checksum = ad77681_compute_xor(checksum_buf, checksum_length, INITIAL_CRC_XOR);
+		
+		
+		if (checksum != checksum_byte)	
+			ret = -1;
+		
+#ifdef CRC_DEBUG
+		
+		char ok[3] = { 'O', 'K' }, fault[6] = { 'F', 'A', 'U', 'L', 'T' };
+		sprintf(print_buf, "\n%x\t%x\t%x\tCRC %s", checksum_byte, checksum, status_byte, ((ret == -1) ? (fault) : (ok)));
+		print_debug(print_buf);	
+		
+#endif // CRC_DEBUG
+		
+	}	
+	
+	return ret;
+}
+
+/**
+ * Conversion from measured data to voltage
+ * @param dev - The device structure.
+ * @param measured_data - structure carying ADC measurement info
+ * @param voltage		- Array of voltage values
+ * @return 0 in case of success, negative error code otherwise.
+ */
+int32_t ad77681_data_to_voltage(struct ad77681_dev *dev,
+	struct adc_data *measured_data,
+	double *voltage)
+{
+	int32_t converted_data = 0;
+	
+	for (measured_data->count = 0; measured_data->count < measured_data->samples; measured_data->count++)
+	{
```

Comment:
as mentioned above for comments, try to fit the code into a maximum of 80 characters per line.

check for similar issues throughout the entire code.